Source/Graphic/HCPath.c

```diff
@@ -230,21 +205,362 @@ HCPathElement HCPathElementAt(HCPathRef self, HCInteger elementIndex) {
+    HCPathElement* elements = (HCPathElement*)HCDataBytes(self->elementData);
+    HCPathRef contourPath = HCPathCreateWithElements(elements + contourStartIndex, contourEndIndex - contourStartIndex);
+    return contourPath;
+}
+
+HCListRef HCPathContourPathsRetained(HCPathRef self) {
+    HCListRef contourPaths = HCListCreate();
+    HCInteger elementCount = HCPathElementCount(self);
+    HCInteger elementIndex = 0;
+    while (elementIndex < elementCount) {
+        HCPathRef contourPath = HCPathContourPathContainingElementRetained(self, elementIndex, NULL, &elementIndex, NULL);
+        if (!HCPathIsEmpty(contourPath)) {
+            HCListAddObjectReleased(contourPaths, contourPath);
+        }
+    }
+    return contourPaths;
+}
+
+HCListRef HCPathOpenContourPathsRetained(HCPathRef self) {
+    HCListRef openContourPaths = HCListCreate();
+    HCInteger elementCount = HCPathElementCount(self);
+    HCInteger elementIndex = 0;
+    while (elementIndex < elementCount) {
+        // Find the contour start, end, and openness
+        HCInteger startIndex;
+        HCInteger endIndex;
+        HCBoolean closed = HCPathContourContainingElementIsClosed(self, elementIndex, &startIndex, &endIndex);
+        elementIndex = endIndex;
+        
+        // If open, create the contour path with the elements from start to end
+        if (!closed) {
+            HCPathElement* elements = (HCPathElement*)HCDataBytes(self->elementData);
+            HCPathRef contourPath = HCPathCreateWithElements(elements + startIndex, endIndex - startIndex);
+            if (!HCPathIsEmpty(contourPath)) {
+                HCListAddObjectReleased(openContourPaths, contourPath);
+            }
+        }
+    }
+    return openContourPaths;
+}
+
+HCListRef HCPathClosedContourPathsRetained(HCPathRef self) {
+    HCListRef closedContourPaths = HCListCreate();
+    HCInteger elementCount = HCPathElementCount(self);
+    HCInteger elementIndex = 0;
+    while (elementIndex < elementCount) {
+        // Find the contour start, end, and openness
+        HCInteger startIndex;
+        HCInteger endIndex;
+        HCBoolean closed = HCPathContourContainingElementIsClosed(self, elementIndex, &startIndex, &endIndex);
+        elementIndex = endIndex;
+        
+        // If closed, create the contour path with the elements from start to end
+        if (closed) {
+            HCPathElement* elements = (HCPathElement*)HCDataBytes(self->elementData);
+            HCPathRef contourPath = HCPathCreateWithElements(elements + startIndex, endIndex - startIndex);
+            if (!HCPathIsEmpty(contourPath)) {
+                HCListAddObjectReleased(closedContourPaths, contourPath);
+            }
+        }
+    }
+    return closedContourPaths;
+}
+
+HCPathRef HCPathOpenContoursAsPathRetained(HCPathRef self) {
+    HCListRef openContours = HCPathOpenContourPathsRetained(self);
+    HCPathRef openPath = HCPathCreateWithSubpaths(openContours);
+    HCRelease(openContours);
```

Comment:
Does it make sense to have a `HCPathCreate` that takes ownership of the given contours?

Reply:
It does, but construction with path elements of the same mechanism hasn't been implemented either. Nor efficient transformation of paths using a transform matrix. All of these should probably be done at the same time.